scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -398,10 +398,11 @@ class FossId internal constructor(
         url: String,
         revision: String? = null
     ): List<Scan> = filter {
+        val isArchived = it.isArchived ?: false
         // The scans in the server contain the url with the credentials so we have to remove it for the
         // comparison. If we don't, the scans won't be matched if the password changes!
         val urlWithoutCredentials = it.gitRepoUrl?.replaceCredentialsInUri()
-        urlWithoutCredentials == url && (revision == null || (it.gitBranch == revision))
+        !isArchived && urlWithoutCredentials == url && (revision == null || (it.gitBranch == revision))
```

Comment:
Unrelated, but mind adding another commit that removes the superfluous parentheses around `(it.gitBranch == revision)`?